libs/nss.c

```diff
@@ -64,6 +89,10 @@ int _nss_tcb_getspnam_r(const char *name, struct spwd *__result_buf,
 	char *file;
 	int retval, saved_errno;
 
+	/* Disallow potentially-malicious user names */
+	if (!is_valid_username(name))
+		return NSS_STATUS_NOTFOUND;
```

Comment:
Should we set `errno`?

Also, after our call to `tcb_safe_open`, should the existing return statement maybe be changed to `return errno == ENOENT ? NSS_STATUS_NOTFOUND : NSS_STATUS_UNAVAIL`? That would be consistent with our handling of `errno` after our call to `fgetspent_r`.

Reply:
Adapted.